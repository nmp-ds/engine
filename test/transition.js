import { setup } from './_helpers.js';
import { expect, test } from 'vitest';
import { durationBase } from '#theme';

setup();

test('transition', async ({ uno }) => {
  const classes = ['none', 'all', 'colors', 'opacity', 'shadow', 'transform'].map(e => `transition-${e}`);
  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});

test('duration', async ({ uno }) => {
  const classes = durationBase.map(e => `duration-${e}`);
  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});

test('delay', async ({ uno }) => {
  const classes = durationBase.map(e => `delay-${e}`);
  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});

test('ease', async ({ uno }) => {
  const classes = ['linear', 'in', 'out', 'in-out'].map(e => `ease-${e}`);
  const { css } = await uno.generate(classes);
  expect(css).toMatchSnapshot();
});
