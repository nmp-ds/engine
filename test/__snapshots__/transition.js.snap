// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`delay 1`] = `
"/* layer: default */
.delay-100{transition-delay:100ms;}
.delay-1000{transition-delay:1000ms;}
.delay-150{transition-delay:150ms;}
.delay-200{transition-delay:200ms;}
.delay-300{transition-delay:300ms;}
.delay-500{transition-delay:500ms;}
.delay-700{transition-delay:700ms;}
.delay-75{transition-delay:75ms;}"
`;

exports[`duration 1`] = `
"/* layer: default */
.duration-100{transition-duration:100ms;}
.duration-1000{transition-duration:1000ms;}
.duration-150{transition-duration:150ms;}
.duration-200{transition-duration:200ms;}
.duration-300{transition-duration:300ms;}
.duration-500{transition-duration:500ms;}
.duration-700{transition-duration:700ms;}
.duration-75{transition-duration:75ms;}"
`;

exports[`ease 1`] = `
"/* layer: default */
.ease-in{transition-timing-function:cubic-bezier(0.4, 0, 1, 1);}
.ease-in-out{transition-timing-function:cubic-bezier(0.4, 0, 0.2, 1);}
.ease-linear{transition-timing-function:linear;}
.ease-out{transition-timing-function:cubic-bezier(0, 0, 0.2, 1);}"
`;

exports[`transition 1`] = `
"/* layer: default */
.transition-all{transition-property:all;transition-timing-function:cubic-bezier(0.4, 0, 0.2, 1);transition-duration:150ms;}
.transition-colors{transition-property:color,background-color,border-color,text-decoration-color,fill,stroke;transition-timing-function:cubic-bezier(0.4, 0, 0.2, 1);transition-duration:150ms;}
.transition-opacity{transition-property:opacity;transition-timing-function:cubic-bezier(0.4, 0, 0.2, 1);transition-duration:150ms;}
.transition-shadow{transition-property:box-shadow;transition-timing-function:cubic-bezier(0.4, 0, 0.2, 1);transition-duration:150ms;}
.transition-transform{transition-property:transform;transition-timing-function:cubic-bezier(0.4, 0, 0.2, 1);transition-duration:150ms;}
.transition-none{transition:none;}"
`;
